CONTRIBUTING.md

```diff
@@ -87,3 +89,343 @@ generated using the pre-commit hooks for
           git add helm/fiftyone-teams-app/README.md
           git commit -m '<COMMIT_MESSAGE>'
           ```
+
+## Localized Deployments (for internal-testing)
+
+### minikube
+
+[minikube](https://minikube.sigs.k8s.io/docs/)
+provides a local kubernetes cluster
+in VMs (or docker containers) on macOS, Linux and Windows.
+
+```shell
+    minikube start
+```
+
+We use
+[Skaffold](https://skaffold.dev/)
+to deploy our application to the minikube cluster with
+Helm and overrides (`values.yaml`).
+When debugging, it may be helpful to start minikube with the flag
+`--keep-running-on-failure` so that the k8s resources are not deleted
+if the helm installation(s) fail.
+
+```shell
+skaffold dev --keep-running-on-failure
+```
+
+It takes a few minutes for the deployments to stabilize as
+we wait for Helm to install MongoDB and cert-managed (for self-signed certificates).
+The fiftyone-teams app installation also takes a few minutes.
+The fiftyone-app will start and upgrade the database
+(because `FIFTYONE_DATABASE_ADMIN: true`)
+and the teams-api will connect to and configure MongoDB.
+
+We use Skaffold "profiles" to control "modules".
+By default, Skaffold will Helm install
+
+* MongoDB
+* cert-manager
+  * CRDs
+  * self-singed ClusterIssuer
+  * cert-manager from chart defaults
+* FiftyOne Teams
+
+To skip installing MongoDB, run
+
+```shell
+skaffold dev --profile no-mongodb
+```
+
+To skip installing cert-manager, run
+
+```shell
+skaffold dev ---profile no-cert-manager
+```
+
+To skip installing both MongoDB and cert-manager, run
+
+```shell
+skaffold dev --profile only-fiftyone
+```
+
+#### Container Images stored in Private Repositories
+
+Our FiftyOne Teams container images are stored in the private repositories
+
+* Docker Hub
+  * Contains released versions
+* Google Artifact Repository (Docker)
+  * Contains private development images created by our
+    [Google Cloud Build](https://github.com/voxel51/cloud-build-and-deploy/)
+    CI/CD runs
+    * Development
+    * Release Candidates
+
+Accessing images in a private repository requires setting
+up authentication to that container registry.
+
+##### Docker Hub
+
+To run released images from Docker hub, configure minikube and Skaffold
+
+1. Start minikube and enable the addon `registry-creds`
+
+    ```shell
+    minikube start
+    minikube addons configure registry-creds
+    ```
+
+1. Create the file `voxel51-docker.json` file
+    1. Get base64 encoded string of docker username and
+       Docker Personal Access Token (PAT)
+
+        ```shell
+        echo -n 'voxeldocker:<YOUR_DOCKER_PERSONAL_ACCESS_TOKEN>' | base64
+        ```
```

Comment:
Wouldn't we need to create PATs for everyone?

Would a shared Voxel51 PAT be easier?

The only folks with dockerhub logins right now are you and me.

Reply:
If a Voxel51 developer needed to access the "published" assets from DockerHub, they would need a PAT.  Instead of using a published asset, they could instead use the private asset sourced from GCP Artifacts instead.

The decision on individual or shared depends on how we want to manage it.  Typically shared credentials violate policies and make for rotation upon employee separation.  Individual credentials are more onerous to manage, and are isolated upon employee separation.